dawn/main/ansible/Ansible.js

```diff
@@ -177,28 +180,29 @@ class SendSocket {
 }
 
 class TCPSocket {
-  constructor(socket) {
+  constructor(socket, logger) {
     this.waitRuntimeConfirm = this.waitRuntimeConfirm.bind(this);
     this.tryUpload = this.tryUpload.bind(this);
+    this.logger = logger;
```

Comment:
not exactly sure what binding does; does it mean the bound function can start being scheduled? if so, we may want `this.logger` to be set before bidning.